ItemCard.tsx

```diff
@@ -0,0 +1,42 @@
+import { View, Image, Text, TouchableOpacity, StyleSheet } from 'react-native';
+
+export interface ImageData {
+    id: number;
+    url: string;
+    width: number;
+    height: number;
+}
+
+interface ItemCardProps extends ImageData {
+    isSelected: boolean;
+}
+
+export const ItemCard = ({ url }: ImageData) => {
```

Comment:
Your props should have a type co-located with your file. `ImageData` is not what we would want to use here.
```
interface Props {
    url: string;
}

export const ItemCard = ({ url }: Props) => {
//other code
}
```